src/Opdex.Platform.WebApi/Caching/BlockBasedCacheControlFilter.cs

```diff
@@ -0,0 +1,54 @@
+using MediatR;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Net.Http.Headers;
+using Opdex.Platform.Application.Abstractions.EntryQueries.Indexer;
+using Opdex.Platform.Application.Abstractions.Queries.Blocks;
+using Opdex.Platform.Domain.Models;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Opdex.Platform.WebApi.Caching;
+
+internal class BlockBasedCacheControlFilter : IAsyncActionFilter
+{
+    private const int BlockIntervalSeconds = 16;
+    private readonly CacheControlOptions _options;
+    private readonly IMediator _mediator;
+
+    public BlockBasedCacheControlFilter(CacheControlOptions options, IMediator mediator)
+    {
+        _options = options;
+        _mediator = mediator;
+    }
+
+    private string CacheTypeDirective => _options.CacheType == CacheType.Public ? "public" : "private";
+
+    public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+    {
+        var headers = context.HttpContext.Response.Headers;
+
+        // clear headers
+        headers.Remove(HeaderNames.CacheControl);
+        headers.Remove(HeaderNames.Pragma);
+
+        var status = await _mediator.Send(new GetIndexerStatusQuery(), CancellationToken.None);
+        if (status.Locked && status.Reason == IndexLockReason.Rewinding)
```

Comment:
we may not want to cache when locked in general, think of if we pause indexing for a release. When its enabled, we might cache old results maybe?

Reply:
Good point, removed the reason check